src/caret_analyze/infra/lttng/records_provider_lttng.py

```diff
@@ -1134,14 +1134,18 @@ def _validate(
         tail_callback = node_path.callbacks[-1]
 
         if node_path.publish_topic_name is not None and \
-            tail_callback.publish_topic_names is not None and \
-            len(tail_callback.publish_topic_names) != 0 and \
-                node_path.publish_topic_name not in tail_callback.publish_topic_names:
-            raise UnsupportedNodeRecordsError('')
+                tail_callback.publish_topics is not None:
+            b_val = False
+            for topics in tail_callback.publish_topics:
+                if node_path.publish_topic_name == topics.topic_name:
+                    b_val = True
+            if b_val is not True:
+                raise UnsupportedNodeRecordsError('')
```

Comment:
- I suggest changing `b_val` to `is_valid` to represent its meaning.
- I suggest changing `topics` to `topic` because it refers to a single data.
- Please add comparison of construction_order.

```suggestion
        if node_path.publish_topic_name is not None and \
                tail_callback.publish_topics is not None:
            is_valid = False
            for topic in tail_callback.publish_topics:
                if node_path.publish_topic_name == topic.topic_name and \
                   node_path.publisher_construction_order == topic.construction_order:
                    is_valid = True
            if is_valid is not True:
                raise UnsupportedNodeRecordsError('')
```